### ISSUE TITLE
Generated snippets with special characters are not correctly escaped.

### ISSUE THREAD

**Issue by hobdell-SAP** [cucumber/cucumber-js#1324]
# Prerequisite Due Diligence
* I could not find related issues about escaping special characters in the error output. 
* I'm only parsing features and step defs (no tests are yet implemented), so definitely cucumber-js.
* This is a bug report. Version Info and steps to reproduce follow. 

# Version Info
```
$ node -v
v12.10.0
$ npm ls cucumber
obfuscated-package@1.0.0 /Users/xxxx/projects/obfuscated-package
└── cucumber@6.0.5
$ set | grep OSTYPE
OSTYPE=darwin19.3.0
(macOS Catalina 10.15.5)
$ bash --version
GNU bash, version 5.0.16(1)-release (x86_64-apple-darwin19.3.0)
(installed with Homebrew)
```

# Steps to reproduce
I'm going from BDD to code. The first step when adding new features is to identify the step defs. I do this by running cucumber, then script-processing the output to extract the missing fragments into stepdef files.

A typical missing step def results in an error like this:
```
? When the user (with permissions) executes the action
        Undefined. Implement with the following snippet:

          When('the user \(with permissions) executes the action', function () {
            // Write code here that turns the phrase above into concrete actions
            return 'pending';
          });
```

After I run my script to extract the snippets, I have a stepdef file with the snippet contents:
```
const assert = require('assert');
const { Given, When, Then } = require('cucumber');
          When('the user \(with permissions) executes the action', function () {
            // Write code here that turns the phrase above into concrete actions
            return 'pending';
          });
```

But running cucumber again still reports the Undefined stepdef.

## Workaround
My script replaces the '`\`' with '`\\`', but the suggestion output should be correct, and not open to interpretation. 

I tracked it down to https://github.com/cucumber/cucumber-js/tree/master/src/formatter/step_definition_snippet_builder before getting lost in the multi-project structure...

**Comment by hobdell-SAP** [comment]
I have edited the description twice to format the backslashes correctly. It seems that escaping of backslashes isn't just a cucumber issue! :)

**Comment by davidjgoss** [comment]
This has now been released in [v8.1.0](https://github.com/cucumber/cucumber-js/releases/tag/v8.1.0) on npm

https://www.npmjs.com/package/@cucumber/cucumber
